bug: Rust edition resolution

### Neovim version (nvim -v)

NVIM v0.12.0-dev-1293+gb256e0476b

### Operating system/version

Ubuntu Ubuntu 24.04.3 LTS

### Read debugging tips

- [x] I have read through the [debugging tips](https://github.com/stevearc/conform.nvim/blob/master/doc/debugging.md).

### Add the debug logs

- [x] I have set `log_level = vim.log.levels.DEBUG` and pasted the log contents below.

### Log file

```console
Log file: /home/roy/.local/state/nvim/conform.log
          2025-09-30 16:32:45[DEBUG] Run CWD: /home/roy/.dotfiles/nvim/.config/nvim
          2025-09-30 16:32:45[DEBUG] stylua exited with code 0
          2025-09-30 16:33:08[DEBUG] Running formatters on /tmp/repro/src/lib.rs: { "rustfmt" }
          2025-09-30 16:33:08[INFO] Run rustfmt on /tmp/repro/src/lib.rs
          2025-09-30 16:33:08[DEBUG] Run command: { "/home/roy/.cargo/bin/rustfmt", "--emit=stdout", "--edition=2021" } <- Incorrect edition
          2025-09-30 16:33:08[DEBUG] Run default CWD: /tmp/repro
          2025-09-30 16:33:08[INFO] rustfmt exited with code 1
          2025-09-30 16:33:08[DEBUG] rustfmt stdout: { "" }
          2025-09-30 16:33:08[DEBUG] rustfmt stderr: { "error: let chains are only allowed in Rust 2024 or later", " --> <stdin>:4:12", "  |", '4 |         if let Some(msg) = opt_msg && msg == "foo"', "  |            ^^^^^^^^^^^^^^^^^^^^^^^", "", "" }
          2025-09-30 16:33:08[ERROR] Formatter 'rustfmt' error: error: let chains are only allowed in Rust 2024 or later
           --> <stdin>:4:12
            |
          4 |         if let Some(msg) = opt_msg && msg == "foo"
            |            ^^^^^^^^^^^^^^^^^^^^^^^
```

### Describe the bug

`rustfmt` is no longer passed the correct edition flag. Seems like #747  is incorrectly resolving the specified edition in non-workspace crates.

### What is the severity of this bug?

breaking (some functionality is broken)

### Steps To Reproduce

1. Create a new rust crate (ensure you have latest toolchain installed).
```bash
rustfmt --version # rustfmt 1.8.0-stable (1159e78c47 2025-09-14)
cd /tmp && cargo new --lib repro && cd ./repro
```

2. Open `Cargo.toml` in nvim using the modified `repro.lua` from the `Minimal init.lua` section below
```bash
nvim -u path/to/repro.lua ./Cargo.toml
```
`Cargo.toml` should look like this
```toml
[package]
name = "repro"
version = "0.1.0"
edition = "2024" # ensure `edition = 2024`

[dependencies]
```

3. Open `src/lib.rs` 
```
:e ./src/lib.rs
```

4. Delete the contents of `lib.rs` and replace it with the following:
```rust
pub fn let_chain() {                                                                                                                                                                                                                                            
    let opt_msg = Some("foo");                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                
        if let Some(msg) = opt_msg && msg == "foo"                                                                                                                                                                                                              
    {                                                                                                                                                                                                                                                           
        println!("{msg}")                                                                                                                                                                                                                                       
    }                                                                                                                                                                                                                                                           
}
```
5. Save the file `:w`
```console
Formatter failed. See :ConformInfo for details
```

### Expected Behavior

`rustfmt` should be passed `--edition=2024` and format the file. Instead `rustfmt` is passed `--edition=2021`. Conform was previously resolving the rust edition properly and would correctly format the example file (even in workspaces).

### Minimal example file

```rust
pub fn let_chain() {
    let opt_msg = Some("foo");

        if let Some(msg) = opt_msg && msg == "foo"
    {
        println!("{msg}")
    }
}
```

### Minimal init.lua

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
	vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({
		"git",
		"clone",
		"--filter=blob:none",
		"--single-branch",
		"https://github.com/folke/lazy.nvim.git",
		lazypath,
	})
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
	"folke/tokyonight.nvim",
	{
		"stevearc/conform.nvim",
		config = function()
			require("conform").setup({
				format_on_save = { timeout_ms = 1000, lsp_format = "fallback" },
				log_level = vim.log.levels.DEBUG,
				formatters_by_ft = {
					rust = { "rustfmt" },
				},
			})
		end,
	},
}
require("lazy").setup(plugins, {
	root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
```

### Additional context

It might make sense to roll back #747. If #747 fixes workspace edition resolution at the expense of breaking non-workspace edition resolution then maybe it needs more testing before publication.

It's also worth noting that I have not encountered the issue #747 was attempting to fix.

--- a/lua/conform/util.lua
+++ b/lua/conform/util.lua
@@ -194,9 +194,9 @@ M.parse_rust_edition = function(dir)
   for _, manifest in ipairs(manifest_files) do
     for line in io.lines(manifest) do
       -- if a project is part of a workspace, the edition might be defined top-level
-      local is_in_workspace = line:match("edition *= *{ *workspace *= *true *}")
-        or line:match("edition.workspace *= *true")
-      local edition = line:match("^edition *= *(%d+)")
+      local is_in_workspace = line:match("^edition *= *{ *workspace *= *true *}")
+        or line:match("^edition.workspace *= *true")
+      local edition = line:match('^edition *= *"(%d+)"')
       if is_in_workspace then
         break
       elseif edition then

